hermes/file_gen_test.go

```diff
@@ -47,13 +28,32 @@ func TestChecksum(t *testing.T) {
 	}
 }
 
-// Test function for the generateFileName function in file_gen.go
+type FileNameTestUnit struct {
+	id       int
+	checksum string
+	want     string
+}
+
+var test_unit_table [5]FileNameTestUnit
+
 func TestFileName(t *testing.T) {
```

Comment:
You also may want to add some tests for your edge cases. Like adding file id= 0, 49, 50, and some other number outside the supported range. We want to ensure that the special files are created properly.